presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -563,30 +633,106 @@ public void setJmxExport(boolean export)
         jmxExportListener.accept(this, export);
     }
 
-    public InternalResourceGroup getOrCreateSubGroup(String name)
+    @Override
+    public int getVersion()
+    {
+        return version;
+    }
+
+    public void setNext(InternalResourceGroup nextNeighbour)
```

Comment:
You can just use `@guardedBy("root")` on the variable.

Reply:
done